test/GameLogic/GameSession.test.ts

```diff
@@ -158,8 +178,27 @@ describe('GameLogic.GameSession', () => {
 		it('Should error when removing recording that is not added', () => {
 			const actionSequence = new ActionSequence();
 
-			const session = new GameSession(() => new Level(levelConfig));
+			const session = new GameSession(new Level(levelConfig));
 			assert.throws(() => session.removeRecording(actionSequence));
 		});
 	});
+
+	describe('levelReset', () => {
+		it('Should use a copy of the level blueprint', () => {
+			//Arrange
+			const level = new Level(levelConfig);
+
+			//Act
+			const session = new GameSession(level);
```

Comment:
You could alternatively just expose `blueprintLevel` as a public field (which I think would be fine, at some point we're likely going to need to access it to view the fresh level in the UI) and then do strict comparison if `blueprintLevel` is not the same as `level`. 

You do the comparison with the level that's passed to `GameSession` but since you clone and assign the clone to the session that assert is never going to be false.